src/btree/row_srch.c

```diff
@@ -614,14 +620,7 @@ leaf_match:	cbt->compare = 0;
 
 	return (0);
 
-err:	/*
-	 * Release the current page if the search started at the root. If the
-	 * search didn't start at the root we should never have gone looking
-	 * beyond the start page.
-	 */
-	WT_ASSERT(session, leaf == NULL || leaf == current);
-	if (leaf == NULL)
-		WT_TRET(__wt_page_release(session, current, 0));
+err:	WT_TRET(__wt_page_release(session, current, 0));
```

Comment:
@agorrod, @michaelcahill: I missed the fact that we went through leaf-only, clearing "current" -- I agree, the test is no longer needed. Sorry, Alex!